code/modules/mob/living/carbon/human/human.dm

```diff
@@ -1145,26 +1145,25 @@ var/list/rank_prefix = list(\
 			C.removed()
 			organs_to_readd += C
 
-	for(var/obj/item/organ/organ in (organs|internal_organs))
-		qdel(organ)
-
 	var/obj/item/weapon/implant/core_implant/CI = get_core_implant()
 	var/checkprefcruciform = FALSE	// To reset the cruciform to original form
 	if(CI)
 		checkprefcruciform = TRUE
 		qdel(CI)
 
-	if(organs.len)
-		organs.Cut()
-	if(internal_organs.len)
-		internal_organs.Cut()
-	if(organs_by_name.len)
-		organs_by_name.Cut()
-	if(internal_organs_by_name.len)
-		internal_organs_by_name.Cut()
-
-
 	if(from_preference)
+		for(var/obj/item/organ/organ in (organs|internal_organs))
+			qdel(organ)
+
+		if(organs.len)
+			organs.Cut()
+		if(internal_organs.len)
+			internal_organs.Cut()
+		if(organs_by_name.len)
+			organs_by_name.Cut()
+		if(internal_organs_by_name.len)
+			internal_organs_by_name.Cut()
+
```

Comment:
use QDEL_NULL_LIST?
(though the more common QDEL_LIST is better and you should update your list and QDEL macros)

Reply:
This is safer to resolve the issue at hand with cruciforms being ejected when the rejuv admin command is used.